samples/samples_for_tutorials/setup/setup_cleanup.js

```diff
@@ -0,0 +1,342 @@
+    brands: ['Google'],
+    fulfillmentInfo,
+    priceInfo: {
+      price: 30.0,
+      originalPrice: 35.5,
+      currency_code: 'USD',
+    },
+    availability: 'IN_STOCK',
+  };
+
+  const retailClient = new ProductServiceClient({ apiEndpoint });
+
+  // Construct request
+  const request = {
+    parent,
+    product,
+    productId,
+  };
+
+  // Run request
+  const response = await retailClient.createProduct(request);
+  console.log(`Product ${response[0].id} created`);
+  return response[0];
+};
+
+const getProduct = async (name) => {
+  const apiEndpoint = 'retail.googleapis.com';
+  const retailClient = new ProductServiceClient({ apiEndpoint });
+
+  // Construct request
+  const request = {
+    name,
+  };
+
+  // Run request
+  const response = await retailClient.getProduct(request);
+  return response;
+};
+
+const deleteProduct = async (name) => {
+  const apiEndpoint = 'retail.googleapis.com';
+  const retailClient = new ProductServiceClient({ apiEndpoint });
+
+  // Construct request
+  const request = {
+    name,
+  };
+
+  // Run request
+  const response = await retailClient.deleteProduct(request);
+  return response;
+};
+
+const deleteProducts = (projectNumber, ids) => {
+  const apiEndpoint = 'retail.googleapis.com';
+  const retailClient = new ProductServiceClient({ apiEndpoint });
+
+  return new Promise(async (resolve, reject) => {
+    try {
+      for (let i = 0; i < ids.length; ++i) {
+        const name = `projects/${projectNumber}/locations/global/catalogs/default_catalog/branches/default_branch/products/${ids[i]}`;
+        await retailClient.deleteProduct({ name });
+      }
+      resolve(true);
+    } catch (err) {
+      reject(err);
+    }
+  });
+};
+
+const getBucketsList = async () => {
+  const storage = new Storage();
+  const [buckets] = await storage.getBuckets();
+  const bucketNames = buckets.map((item) => item.name);
+  console.log(bucketNames);
+  resolve(buckets);
+};
+
+const isBucketExist = async (name) => {
+  const storage = new Storage();
+  const [buckets] = await storage.getBuckets();
+  const bucketNames = buckets.map((item) => item.name);
+  return bucketNames.indexOf(name) !== -1 ? true : false;
+};
+
+const createBucket = (name) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      if (await isBucketExist(name)) {
+        console.log(`Bucket ${name} alreaty exists`);
+        resolve(false);
+      } else {
+        const storage = new Storage();
+        const location = 'us';
+        const storageClass = 'STANDARD';
+        const createdBucket = await storage.createBucket(name, {
+          location,
+          [storageClass]: true,
+        });
+        console.log(`Bucket ${createdBucket[0].name} created.`);
+        resolve(createdBucket);
+      }
+    } catch (error) {
+      reject(error);
+    }
+  });
+};
+
+const deleteBucket = async (name) => {
+  const storage = new Storage();
+  await storage.bucket(name).delete();
```

Comment:
If tests fail, will we start to leak buckets and other resources? BigQuery does a great job of doing this, here:

https://github.com/googleapis/nodejs-bigquery/blob/main/samples/test/datasets.test.js#L124

Basically, if you name resources such that you know they're for these specific tests, and then have a cleanup step that deletes resources older than a few hours, we create tests that won't leak resources over time.

Reply:
There is an environment variable BUCKET_NAME (stored in the secret manager), when the GCS bucket is being created   the same bucket name is used. There is also a check if such bucket is already exists, if so, the products.json will be uploaded to this bucket. The tests will create only one GCS bucket and BigQuery table